### PR TITLE
[KBFS-1773] Fix race with enabling journaling and login

### PR DESCRIPTION
**Pull request by akalin-keybase**
Initialize the journal even if getting the current
user info fails, which is needed to enable existing
journals. That way, when the user does log in, the
existing journals will be enabled then.

### CODE REVIEW

**Comment by akalin-keybase** [comment]
Ah, don't look at this quite yet...

**Comment by akalin-keybase** [comment]
Okay, this is ready for review now.

File: libkbfs/init.go [hunk 1]
```diff
@@ -66,7 +67,7 @@ type InitParams struct {
 	LogFileConfig logger.LogFileConfig
 
 	// TLFJournalBackgroundWorkStatus is the status to use to
-	// pass into config.EnableJournaling. Only has an effect when
+	// pass into journalServer.EnableJournaling. Only has an effect when
```
Comment (strib) [inline libkbfs/init.go]: `journalServer` -> `JournalServer`

File: libkbfs/journal_server.go [hunk 2]
```diff
@@ -130,6 +130,9 @@ func makeJournalServer(
 	bcache BlockCache, dirtyBcache DirtyBlockCache, bserver BlockServer,
 	mdOps MDOps, onBranchChange branchChangeListener,
 	onMDFlush mdFlushListener, diskLimiter diskLimiter) *JournalServer {
+	if len(dir) == 0 {
+		panic("dir unexpectedly empty")
```
Comment (strib) [inline libkbfs/journal_server.go]: This wording is a bit ambiguos, I first thought it meant the directory has no entries in it.  maybe "dir" -> "dir string"?

File: libkbfs/init.go (hunk 1 above)
Comment (akalin-keybase) [inline libkbfs/init.go]: Fixed

File: libkbfs/journal_server.go (hunk 2 above)
Comment (akalin-keybase) [inline libkbfs/journal_server.go]: Fixed

**Pull request merged**